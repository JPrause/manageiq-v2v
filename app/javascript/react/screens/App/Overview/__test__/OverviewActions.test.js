import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { mockRequest, mockReset } from '../../../../../common/mockRequests';

import { fetchTransformationPlansAction, setMigrationsFilterAction } from '../OverviewActions';
import { requestTransformationPlansData } from '../overview.transformationPlans.fixtures';

const middlewares = [thunk, promiseMiddleware()];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('fetchTransformationPlansAction', () => {
  const { fetchTransformationPlansUrl, response } = requestTransformationPlansData;

  test('dispatches PENDING and FULFILLED actions', () => {
    mockRequest({
      method: 'GET',
      url: fetchTransformationPlansUrl,
      params: null,
      status: 200,
      ...response
    });

    return store
      .dispatch(
        fetchTransformationPlansAction({
          url: fetchTransformationPlansUrl,
          archived: false
        })
      )
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      method: 'GET',
      url: fetchTransformationPlansUrl,
      params: null,
      status: 404,
      ...response
    });

    return store
      .dispatch(
        fetchTransformationPlansAction({
          url: fetchTransformationPlansUrl,
          archived: false
        })
      )
      .catch(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});

describe('setMigrationsFilterAction', () => {
  const activeFilter = 'Migrations Plans Not Started';

  test('dispatches an action with correct type and payload', () => {
    store.dispatch(setMigrationsFilterAction(activeFilter));

    expect(store.getActions()).toMatchSnapshot();
  });

  test('dispatches additional actions', () => {
    const SOME_ACTION_TYPE = 'SOME_ACTION_TYPE';
    const payload = 'Some payload';
    store.dispatch(setMigrationsFilterAction(activeFilter, { [SOME_ACTION_TYPE]: payload }));

    expect(store.getActions()).toMatchSnapshot();
  });
});
