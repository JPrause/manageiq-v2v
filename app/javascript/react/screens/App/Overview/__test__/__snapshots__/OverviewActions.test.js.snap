// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchTransformationPlansAction dispatches PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_PENDING",
  },
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_FULFILLED",
  },
]
`;

exports[`setMigrationsFilterAction dispatches additional actions 1`] = `
Array [
  Object {
    "payload": "Migrations Plans Not Started",
    "type": "V2V_SET_MIGRATIONS_FILTER",
  },
  Object {
    "payload": "Some payload",
    "type": "SOME_ACTION_TYPE",
  },
]
`;

exports[`setMigrationsFilterAction dispatches an action with correct type and payload 1`] = `
Array [
  Object {
    "payload": "Migrations Plans Not Started",
    "type": "V2V_SET_MIGRATIONS_FILTER",
  },
]
`;
